lib/middleware/request-handler.ts

```diff
@@ -78,17 +86,15 @@ function validatePollId(session: Session, newPollId: number): { isValid: boolean
     }
 }
 
-function startAllNewBuffers(subscriptionStore: Map<Subscription>): Object[]
+function startAllNewBuffers(subscriptions: Subscription[]): Object[]
 {
     var buffers: Object[] = [];
     // Check if we have new data for any subscriptions
-    var hasNewData: boolean = false;
-    subscriptionStore.forEach((sub: Subscription): boolean => {
-        hasNewData = !sub.buffer.isEmpty();
+    var isAllEmpty: boolean = _.every(subscriptions, (sub: Subscription): boolean => {
         return sub.buffer.isEmpty();
     });
```

Comment:
nit: you could use _.some and return !sub.buffer.isEmpty(). This way you will iterate until the first non empty buffer and it is actually the 'all buffers are not empty' state that is interesting to you, so you won't need to negate the result.